rspotify-http/src/common.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+use std::fmt;
+
+use maybe_async::maybe_async;
+use rspotify_model::ApiError;
+use serde_json::Value;
+
+pub type Headers = HashMap<String, String>;
+pub type Query<'a> = HashMap<&'a str, &'a str>;
+pub type Form<'a> = HashMap<&'a str, &'a str>;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("request unauthorized")]
+    Unauthorized,
+
+    #[error("exceeded request limit")]
+    RateLimited(Option<usize>),
+
+    #[error("request error: {0}")]
+    Request(String),
+
+    #[error("status code {0}: {1}")]
+    StatusCode(u16, String),
+
+    #[error("spotify error: {0}")]
+    Api(#[from] ApiError),
+
+    #[error("input/output error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
```

Comment:
Most libraries also use `Error`, you can then use it as `http::Error` to avoid ambiguity wherever needed. The thing is that we'd have to change `ClientError` to be `rspotify::Error` as well for consistency. But seeing that we have `ApiError` as well it might be more clear by using a prefix. I'll change it to `HttpError`, but if you think the other option is better let me know and I'll undo it.

Reply:
`HttpError` is good enough :)